File: OOPs/Constructors/default_constructor.py
class Book:
    def __init__(self):
        print("Constructor will be called first and memory will be allocated")
        self.x = 0

    def setX(self, x):
        self.x = x

    def getX(self):
        return self.x


if __name__ == "__main__":
    book = Book()
    book.setX(1234)
    print(book.getX())

File: OOPs/Constructors/default_constructor.cpp
#include <bits/stdc++.h>
using namespace std;

class Book {
    public: 
        int x;
    public:
        Book() {
            // Default Constructor No need to write
            cout << "Constructor will be called first and memory will be allocated" << endl;
        }
        void setX(int x) {
            this->x = x;
        }
        int getX() {
            return x;
        }
};

int main() {
    Book book;
    book.setX(1234);
    cout << book.getX() << endl;
    return 0;
}